include/solvers/vof_subequations_interface.h

```diff
@@ -236,23 +339,165 @@ class VOFSubequationsInterface
+   */
+  bool
+  validity_map_has_been_reset()
+  {
+    for (const auto &pair : subequations_solutions_validity)
+      {
+        if (pair.second == true)
+          return false;
+      }
+    return true;
+  }
+
 private:
-  MultiphysicsInterface<dim> *multiphysics_interface;
+  /**
+   * @brief Reset all boolean in the validity associated with active
+   * subequations to @p false.
+   */
+  void
+  reset_subequations_solutions_validity()
+  {
+    for (const VOFSubequationsID &subequation_id : this->active_subequations)
+      {
+        subequations_solutions_validity[subequation_id] = false;
+      }
+  }
 
+  /// Parallel consol output object
   const ConditionalOStream pcout;
 
-  // Data structure that stores all enabled subequations
+  /// VOF DoFHandler associated with solved equations
+  std::optional<std::reference_wrapper<const DoFHandler<dim>>> dof_handler_vof;
+
+  /** Filtered VOF solution field associated with solved equations
+   * @note Used in the computation of VOF phase gradient projection */
+  GlobalVectorType vof_filtered_solution_vector;
+
+  /** VOF solution field associated with solved equations
+   * @note Serves as initial condition in algebraic reinitialization */
+  GlobalVectorType vof_solution_vector;
+
+  /// Data structure that stores all enabled subequations
   std::vector<VOFSubequationsID> active_subequations;
 
-  // Subequations stored within a map of shared pointer to ensure proper
-  // memory management.
+  /** Subequations are stored within a map of shared pointer to ensure proper
+   * memory management */
   std::map<VOFSubequationsID, std::shared_ptr<PhysicsSubequationsSolverBase>>
     subequations;
 
+  /// Map of DoFHandler of subequations
   std::map<VOFSubequationsID, DoFHandler<dim> *> subequations_dof_handler;
 
-  // Present solutions
+  /// Present solutions map
   std::map<VOFSubequationsID, GlobalVectorType *> subequations_solutions;
+
+  /**
+   * Booleans indicating if the subequation has been solved with the set VOF
+   * solution vector
+   *
+   * During initialization
+   * (VOFSubequationsInterface<dim>::initialize_subequations),
+   * all boolean are set to @p false.
+   *
+   * We start by setting a VOF filtered solution and DOFHandler
+   * (VOFSubequationsInterface<dim>::set_vof_filtered_solution_and_dof_handler),
+   * and then start solving equations making sure that dependencies are met
+   * (VOFPhaseGradientProjection<dim>::check_dependencies_validity,
+   * VOFCurvatureProjection<dim>::check_dependencies_validity, and
+   * VOFAlgebraicInterfaceReinitialization<dim>::check_dependencies_validity).
+   * For instance, to solve the curvature projection subequation, the phase
+   * gradient projection subequation has to be solved before as the solution is
+   * used in the subequation.
+   *
+   * A subequation in the validity map is to @p true only when it is solved and
+   * remains that way until a new VOF filtered solution and DOFHandler are set.
```

Comment:
```suggestion
   * A subequation in the validity map is to @p true only when it is solved and
   * it remains valid until a new VOF filtered solution or DOFHandler are set.
```
maybe specify the function name which do that setting.